File: BOJ/2798_블랙잭/go/Main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {

	var n, m int
	fmt.Scan(&n, &m)
	cards := make([]int, n)

	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	text := scanner.Text()
	for i, v := range strings.Split(text, " ") {
		cards[i], _ = strconv.Atoi(v)
	}

	max := 0
	for i := 0; i < n; i++ {
		for j := i + 1; j < n; j++ {
			for k := j + 1; k < n; k++ {
				tmpSum := cards[i] + cards[j] + cards[k]
				if tmpSum <= m && tmpSum > max{
					max = tmpSum
				}
			}
		}
	}
	fmt.Print(max)
}

File: BOJ/2798_블랙잭/java/Main.java

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String[] nm = scanner.nextLine().split(" ");

        int n =  Integer.parseInt(nm[0]);
        int m =  Integer.parseInt(nm[1]);

        int[] card = new int[m];
        String[] cardValue = scanner.nextLine().split(" ");
        for (int i = 0; i < n; i++) {
            card[i] = Integer.parseInt(cardValue[i]);
        }

        int max = 0;
        int tmpSum;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                for (int k = j + 1; k < n; k++) {
                    tmpSum = card[i] + card[j] + card[k];
                    if (tmpSum <= m && tmpSum > max) {
                        max = tmpSum;
                    }
                }
            }
        }
        System.out.println(max);
    }
}

